### PR TITLE
Updates to docstrings

### PR DESCRIPTION
**Pull request by hoodmane**
Increased the number of cross references, removed some types for better compatibility with sphinx-autodoc-typehints and made various other minor improvements.

### CODE REVIEW

File: src/py/_pyodide/_core_docs.py [hunk 1]
```diff
@@ -1005,9 +991,7 @@ def create_proxy(
 
         .. code-block:: python
 
-            from js import o
-            d = {}
-            o.d = create_proxy(d, roundtrip=True)
+            from js import o d = {} o.d = create_proxy(d, roundtrip=True)
```
Comment (ryanking13) [inline src/py/_pyodide/_core_docs.py:994]:
```suggestion
            from js import o
            d = {}
            o.d = create_proxy(d, roundtrip=True)
```

File: src/py/_pyodide/_core_docs.py [hunk 2]
```diff
@@ -1127,75 +1110,77 @@ def to_js(
...
+    In addition to the normal conversions, convert JavaScript `Date
+    <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date>`_
+    objects to :py:class:`~datetime.datetime` objects:
 
-        .. code-block:: python
+    .. code-block:: python
 
-            def default_converter(_value, _ignored1, _ignored2):
-                raise Exception("Failed to completely convert object")
+        from datetime import datetime from js import Date def
```
Comment (ryanking13) [inline src/py/_pyodide/_core_docs.py:1126]:
```suggestion
        from datetime import datetime
        from js import Date
```

File: src/py/_pyodide/_core_docs.py [hunk 3]
```diff
@@ -1127,75 +1110,77 @@ def to_js(
...
+        default_converter(value, _ignored1, _ignored2):
```
Comment (ryanking13) [inline src/py/_pyodide/_core_docs.py:1127]:
```suggestion
        def default_converter(value, _ignored1, _ignored2):
```

File: src/py/_pyodide/_core_docs.py [hunk 4]
```diff
@@ -1127,75 +1110,77 @@ def to_js(
...
-        referential pair:
+    We can use the following ``default_converter`` to convert ``Pair`` to
+    ``Array``:
 
-        .. code-block:: javascript
+    .. code-block:: python
 
-            p = Pair(0, 0);
-            p.first = p;
+        from js import Array def default_converter(value, convert, cache):
```
Comment (ryanking13) [inline src/py/_pyodide/_core_docs.py:1156]:
```suggestion
        from js import Array
        
        def default_converter(value, convert, cache):
```

**Review by ryanking13 — APPROVED** [review]
Thanks for the documentation work @hoodmane!

It seems like the auto-formatter broke some of the codeblocks, otherwise looks good to me.

**Comment by hoodmane** [comment]
Thanks @ryanking13 I always have this problem.

**Pull request merged**